ui/src/testing/utils.js

```diff
@@ -17,3 +17,26 @@ export const compareJSX = (actual, expected) => {
     .debug();
   expect(actualOutput).toBe(expectedOutput);
 };
+
+/**
+ * Replace objects in an array with objects that have new values, given a match
+ * criteria.
+ * @param {Array} array - Array to be reduced.
+ * @param {String} key - Object key to compare the match criteria e.g. "name".
+ * @param {String} match - Match criteria e.g. "Bob".
+ * @param {Object} newValues - Values to insert or update in the object.
+ * @returns {Array} The reduced array.
+ */
+export const reduceInitialState = (array, key, match, newValues) => {
```

Comment:
@squidsoup @huwshimi as I was writing the tests for Storage I was wondering - do we want a file somewhere that just has all the initial state you would expect from the app in steady-state (i.e. everything has loaded)? Then we could import it into any connected component test file, and use a simple function like this to mock the small pieces of state differences that we're interested in. Any thoughts?

Reply:
From memory crbs has one of those and it seemed helpful.

Reply:
Yep we use [this](https://github.com/canonical-web-and-design/crbs-ui/blob/master/src/testing/default-state.js) in rbac, and do something like this to ensure defaultState isn't mutated:

```javascript
  beforeEach(() => {
    jest.resetModules();
    // eslint-disable-next-line global-require
    state = require("../../../testing/default-state").default;
  });
```

Reply:
Ah, nice! I didn't realise you'd already done that.